src/p2p_plugin.c

```diff
@@ -235,14 +244,13 @@ ncclResult_t nccl_p2p_ib_init(int *num_devs, nccl_ib_dev_t *ncclIbDevs, char *nc
             ncclIbDevs[ncclNIbDevs].maxQp = ncclParamSharpMaxComms();
             ncclNSharpDevs++;
           }
+          pthread_create(ncclIbAsyncThread, NULL, ncclIbAsyncThreadMain, context);
+          pthread_detach(*ncclIbAsyncThread); // will not be pthread_join()'d
+          ncclSetThreadName(*ncclIbAsyncThread, "NCCL IbAsync %2d", ncclNIbDevs);
```

Comment:
We recently fixed a SEGV bug with ncclSetThreadName where a thread could exit after the detach but before SetThreadName had completed. See commit d5f6a5f210e980eb06ada5733635baa18ada1f22

Reply:
is this fix in public NCCL github? I have taken changes from public github

Reply:
i will take those changes when it is public